From 5119f0ec0df2d456031a26c7d7c8849f8cfbf648 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Thu, 19 Jul 2018 11:27:12 +0300
Subject: [PATCH 4171/5242] MLK-18978: drm: imx: dcss: fix warning message
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

commit  2f2ae3ea0348a45ee0fddea0dbf53f43ae9d9a40 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch fixes a warning message that's seen during 4.14 compilation.

  CC      drivers/gpu/drm/imx/dcss/dcss-plane.o
drivers/gpu/drm/imx/dcss/dcss-plane.c: In function ‘dcss_plane_atomic_set_base’:
drivers/gpu/drm/imx/dcss/dcss-plane.c:347:4: warning: ‘caddr’ may be used uninitialized in this function [-Wmaybe-uninitialized]
    dcss_dec400d_addr_set(dcss_plane->dcss, p1_ba, caddr);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The warning appeared because of a logic error in code that was
introduced by the following commit:

commit a46e1cd7df28 ("MGS-3560 [#imx-913] Enable DRM compression for
mscale board")

The above commit moved a piece of code, that was located under
switch-case branch, under a do-while() loop and converted all the
returns to breaks. However, a break makes the code exit the while loop
and will continue executing the code under the case branch. Hence, caddr
may be used whithout being initialized.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
CC: Yong Gan <yong.gan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dcss/dcss-plane.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/imx/dcss/dcss-plane.c b/drivers/gpu/drm/imx/dcss/dcss-plane.c
index 074acc6..0c19da0 100644
--- a/drivers/gpu/drm/imx/dcss/dcss-plane.c
+++ b/drivers/gpu/drm/imx/dcss/dcss-plane.c
@@ -324,14 +324,14 @@ static void dcss_plane_atomic_set_base(struct dcss_plane *dcss_plane)
 
 				mdata = dma_buf->priv;
 				if (!mdata || mdata->magic != VIV_VIDMEM_METADATA_MAGIC) {
-					break;
+					return;
 				}
 				compressed = mdata->compressed ? true : false;
 				compressed_format = mdata->compress_format;
 
 				gem_obj = dcss_plane_gem_import(plane->dev, mdata->ts_dma_buf);
 				if (IS_ERR(gem_obj)) {
-					break;
+					return;
 				}
 
 				caddr = to_drm_gem_cma_obj(gem_obj)->paddr;
-- 
1.7.9.5

